.github/docker/linux/pr_build.sh

```diff
@@ -0,0 +1,186 @@
+#!/bin/bash
+
+#
+# Copyright 2020 New Relic Corporation. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+
+#
+# Build the daemon, agent and axiom tests in accordance  with our pull request
+# guidelines, which are documented in the README.md file. Additionally, run
+# the axiom tests.
+#
+# This script is meant to run in GHA  with the GitHub Pull Request
+# action which  handles reporting the outcome to
+# GitHub for us.
+#
+
+set -e
+set -u
+
+die() {
+  echo
+  echo >&2 "FATAL: $*"
+  echo
+  exit 1
+}
+
+#
+# Ensure /usr/local/bin should be in the PATH.
+#
+case ":$PATH:" in
+  *:/usr/local/bin:*) ;;
+  *) PATH=/usr/local/bin:$PATH
+esac
+
+case ":$PATH:" in
+  *:/usr/local/go/bin:*) ;;
+  *) PATH=/usr/local/go/bin:$PATH
+esac
+
+export PATH
+
+
+#
+# Get PHPS from the environment.
+#
+
+PHPS=${PHP_VER}
+
+printf \\n
+printf 'running daemon tests\n'
+make -r -s daemon daemon_integration  "ARCH=${ARCH}"
+
+
+#
+# Limit valgrind to just the Linux builders. On Alpine Linux, valgrind
+# appears to alter the results of floating point to string conversions
+# causing spurious test failures.
+#
+
+PHP_SAPIS=$(php-config --php-sapis)
+
+case $PHP_SAPIS in
+  *embed*)
+     PHP_SAPIS_EMBED=1
+    ;;
+  *)
+    PHP_SAPIS_EMBED=0
+    ;;
+esac
+
+#
+# This check can be removed when the valgrind PHP 8.0 mem issues 
+# on 32-bit OSs are resolved.  64-bit has different issue.
+#
+VALGRIND_ISSUE=0
+    echo "php = $PHPS"
+case $PHPS in
+  *8.0*)
+       VALGRIND_ISSUE=1
+    ;;
+  *)
+    VALGRIND_ISSUE=0
+    ;;
+esac
+    echo "valgrindissue : $VALGRIND_ISSUE"
+
+if [ "$(uname)" = Linux ] && [ ! -e /etc/alpine-release ] && [ $PHP_SAPIS_EMBED ]&& [ $VALGRIND_ISSUE = 0 ]; then
+  do_valgrind=yes
+  printf \\n
+  printf 'grinding axiom tests\n'
+  make -r -s -j $(nproc) axiom-valgrind "ARCH=${ARCH}"
+else
+  do_valgrind=
+  printf \\n
+  printf 'running axiom tests\n'
+  make -r -s -j $(nproc) axiom-run-tests "ARCH=${ARCH}"
+fi
+
+#
+# Run the agent integration tests without requiring any changes to the
+# PHP installation, and (ideally) without the tests being negatively
+# affected by any existing INI settings. For example, a pre-existing
+# "extension = newrelic.so". For each version of PHP, override the INI
+# file, INI directory and the extension directory using environment
+# variables. 
+#
+
+INTEGRATION_DIR="${PWD}/integration.tmp"
+if [ ! -d "$INTEGRATION_DIR" ]; then
+  mkdir "$INTEGRATION_DIR"
+  mkdir "${INTEGRATION_DIR}/etc"
+fi
+
+rm -rf "${INTEGRATION_DIR:?}"/*
```

Comment:
Why is there a `:?` on this line? I'm not familiar with that idiom.

Reply:
It tests for the existence of $INTEGRATION_DIR before trying to remove it.

Much of this script was modified from the original pr build script `build-pull-request.sh`.